src/main/java/uk/gov/ons/ctp/response/action/service/CaseNotificationService.java

```diff
@@ -18,109 +17,141 @@
+ * action.case table.
  */
 @Service
 public class CaseNotificationService {
   private static final Logger log = LoggerFactory.getLogger(CaseNotificationService.class);
 
+  private static final String ACTION_PLAN_NOT_FOUND = "No action plan found with id: %s";
+  private static final String ACTION_CASE_ALREADY_EXISTS = "Action case already exists with id: %s";
+  private static final String ACTION_CASE_NOT_FOUND = "No action case found with id: %s";
   private static final int TRANSACTION_TIMEOUT = 30;
 
-  @Autowired private ActionCaseRepository actionCaseRepo;
+  private final ActionCaseRepository actionCaseRepo;
+  private final ActionPlanRepository actionPlanRepo;
 
-  @Autowired private ActionPlanRepository actionPlanRepo;
+  private final ActionService actionService;
 
-  @Autowired private ActionService actionService;
+  private final CollectionExerciseClientService collectionSvcClientService;
 
-  @Autowired private CollectionExerciseClientService collectionSvcClientServiceImpl;
+  public CaseNotificationService(
+      ActionCaseRepository actionCaseRepo,
+      ActionPlanRepository actionPlanRepo,
+      ActionService actionService,
+      CollectionExerciseClientService collectionSvcClientService) {
+    this.actionCaseRepo = actionCaseRepo;
+    this.actionPlanRepo = actionPlanRepo;
+    this.actionService = actionService;
+    this.collectionSvcClientService = collectionSvcClientService;
+  }
 
   @Transactional(
       propagation = Propagation.REQUIRED,
       readOnly = false,
       timeout = TRANSACTION_TIMEOUT)
   public void acceptNotification(final CaseNotification notification) throws CTPException {
-    final String actionPlanIdStr = notification.getActionPlanId();
-    final UUID actionPlanId = UUID.fromString(actionPlanIdStr);
-    final ActionPlan actionPlan = actionPlanRepo.findById(actionPlanId);
-    final UUID caseId = UUID.fromString(notification.getCaseId());
-    final UUID collectionExerciseId = UUID.fromString(notification.getExerciseId());
-    UUID partyId = null;
-    if (notification.getPartyId() != null) {
-      partyId = UUID.fromString(notification.getPartyId());
+
+    UUID caseId = UUID.fromString(notification.getCaseId());
+
+    switch (notification.getNotificationType()) {
+      case REPLACED:
+      case ACTIVATED:
+        ActionCase actionCase = createActionCase(notification);
+        saveActionCase(actionCase);
+        break;
+
+      case ACTIONPLAN_CHANGED:
+        updateActionCase(caseId, UUID.fromString(notification.getActionPlanId()));
+        break;
+
+      case DISABLED:
+      case DEACTIVATED:
+        actionService.cancelActions(caseId);
+        deleteActionCase(caseId);
+        break;
+
+      default:
+        log.with("notification_type", notification.getNotificationType())
+            .warn("Unknown case notification type", notification.getNotificationType());
+        throw new IllegalArgumentException("Invalid notification type");
```

Comment:
I would log this as `.error` because it will actually block the Rabbit queue until the bad message is DLQ'ed